pandas/core/frame.py

```diff
@@ -5792,6 +5805,8 @@ class    max    type
             new_obj = self
         else:
             new_obj = self.copy()
+        if allow_duplicates is not lib.no_default:
```

Comment:
shouldnt we do the same as you are doing in insert and just set this here e.g.

```

if allow_duplicates is not lib.no_default:
    allow_duplicates = False
.....
```

simpler & more readable

Reply:
Your suggestion has got to be a typo. That change would render the argument always `False` and fail my tests.

If I take out the next line, which is type-checking, then I will have to take out my test https://github.com/johnzangwill/pandas/blob/dddae0734e37ab9fa4ae4a89e68043c4631fd68c/pandas/tests/frame/methods/test_reset_index.py#L474

If I copy in the code from `insert`, then the `lib.no_default` no longer has any function.

I only used` lib.no_default` to avoid your original objections to this argument. I don't like it, because it makes no sense in the documentation and disguises the real default, which is `False`.
I would be more than happy to take it out and go back to the original much simpler `allow_duplicates: bool = False`, as `insert` is in main and as I already have in `Series`: https://github.com/johnzangwill/pandas/blob/dddae0734e37ab9fa4ae4a89e68043c4631fd68c/pandas/core/series.py#L1369

Let me know what you want!